admin-cli/lib/generate-changelog.ts

```diff
@@ -30,17 +30,15 @@ export function generateChangelog(args) {
   createIfMissing(options);
 
   let oldContent = options.dryRun ? '' : fs.readFileSync(options.inFile, 'utf8');
-  let newContent = '';
 
   if (oldContent.indexOf('<a name=') !== -1) {
     oldContent = oldContent.substring(oldContent.indexOf('<a name='));
   }
 
   return conventionalChangelogObservable(options).pipe(
-    tap(buffer => {
-      newContent += buffer.toString();
-    }),
-    finalize(() => {
```

Comment:
@PascalPrecht The thing about `finalize` is that it doesn't quite behave the way we want in this case.

Check this:
https://stackblitz.com/edit/typescript-cpibdy?file=index.ts

Notice that output in the console is `1, z, y, x` which basically means that the last `complete` callback in the registration pipe line is called first. This ruins the fun in our case because we really wanted the `finalize` here to be called first.

So, I went for a different construct using `scan` and `last` which also eliminates the shared state (`newContent`). That slightly changes the semantics because now we have an Observable that will only emit once the entire changelog was consumed but I guess that is fine for what we want.

Reply:
Thank you, this makes sense!